client/my-sites/marketing/traffic/index.js

```diff
@@ -110,7 +110,9 @@ const SiteSettingsTraffic = ( {
 );
 
 const connectComponent = connect( state => ( {
-	isJetpack: isJetpackSite( state, getSelectedSiteId( state ) ),
+	isJetpackAdmin:
+		isJetpackSite( state, getSelectedSiteId( state ) ) &&
+		canCurrentUser( state, getSelectedSiteId( state ), 'manage_options' ),
```

Comment:
Is there a reason why we didn't use the existing `isAdmin` prop? Seems like the `isJetpackAdmin` prop could be just a combination of the existing props: `isJetpack && isAdmin`.

Reply:
That would make sense, but I'm getting an error when trying it. Am I doing something wrong or misunderstanding you?

```jsx
const connectComponent = connect( state => ( {
	isAdmin: canCurrentUser( state, getSelectedSiteId( state ), 'manage_options' ),
	isJetpackAdmin: isJetpackSite( state, getSelectedSiteId( state ) ) && isAdmin,
} ) );
```

<img width="1166" alt="Screenshot 2020-02-19 at 18 51 59" src="https://user-images.githubusercontent.com/43215253/74865620-51e68000-5349-11ea-840d-74204724894c.png">

Reply:
Yeah, that won't work. You're getting an error because you're trying to use something that's not yet defined 😉 

Basically, to make that work, you'll have to change from returning an object:

```
const connectComponent = connect( state => ( {
	isAdmin: canCurrentUser( state, getSelectedSiteId( state ), 'manage_options' ),
	isJetpackAdmin: isJetpackSite( state, getSelectedSiteId( state ) ) && isAdmin,
} ) );
```

to using a function:

```

const connectComponent = connect( state => {
	const siteId = getSelectedSiteId( state );
	const isAdmin = canCurrentUser( state, siteId, 'manage_options' );
	const isJetpack = isJetpackSite( state, siteId );

	return {
		isAdmin,
		isJetpack,
	}
} );
```

and then you could either use `isAdmin && isJetpack` in your code or define another prop `isJetpackAdmin`  and use that instead, based on what makes the most sense in this component.

Let me know if that makes sense 😉

Reply:
Thank you! I think that I've got it working now. :)